services/core-web/common/actionCreators/permitActionCreator.ts

```diff
@@ -0,0 +1,500 @@
+  dispatch(showLoading());
+  return CustomAxios()
+    .delete(
+      `${ENVIRONMENT.apiUrl}${API.PERMIT_DELETE(mineGuid, permitGuid)}`,
+      createRequestHeader()
+    )
+    .then((response) => {
+      notification.success({
+        message: "Successfully deleted permit and all related permit amendments and documents.",
+        duration: 10,
+      });
+      dispatch(success(reducerTypes.DELETE_PERMIT));
+      return response;
+    })
+    .catch((err) => {
+      dispatch(error(reducerTypes.DELETE_PERMIT));
+      throw new Error(err);
+    })
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const deletePermitAmendment =
+  (mineGuid: string, permitGuid: string, permitAmdendmentGuid: string) => (dispatch) => {
+    dispatch(request(reducerTypes.DELETE_PERMIT_AMENDMENT));
+    dispatch(showLoading());
+    return CustomAxios()
+      .delete(
+        `${ENVIRONMENT.apiUrl}${API.PERMIT_AMENDMENT(mineGuid, permitGuid, permitAmdendmentGuid)}`,
+        createRequestHeader()
+      )
+      .then((response) => {
+        notification.success({
+          message: "Successfully deleted permit amendment and all related documents.",
+          duration: 10,
+        });
+        dispatch(success(reducerTypes.DELETE_PERMIT_AMENDMENT));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.DELETE_PERMIT_AMENDMENT));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading()));
+  };
+
+export const fetchPermitConditions = (permitAmdendmentGuid) => (dispatch) => {
+  dispatch(request(reducerTypes.GET_PERMIT_CONDITIONS));
+  dispatch(showLoading());
+  return CustomAxios()
+    .get(
+      `${ENVIRONMENT.apiUrl}${API.PERMIT_CONDITIONS(null, null, permitAmdendmentGuid)}`,
+      createRequestHeader()
+    )
+    .then((response) => {
+      dispatch(success(reducerTypes.GET_PERMIT_CONDITIONS));
+      dispatch(permitActions.storePermitConditions(response.data));
+    })
+    .catch(() => dispatch(error(reducerTypes.GET_PERMIT_CONDITIONS)))
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const createPermitCondition = (permitAmdendmentGuid, payload) => (dispatch) => {
+  dispatch(request(reducerTypes.CREATE_PERMIT_CONDITION));
+  dispatch(showLoading());
+  return CustomAxios()
+    .post(
+      `${ENVIRONMENT.apiUrl}${API.PERMIT_CONDITIONS(null, null, permitAmdendmentGuid)}`,
+      { permit_condition: payload },
+      createRequestHeader()
+    )
+    .then((response) => {
+      notification.success({
+        message: "Successfully created a new condition",
+        duration: 10,
+      });
+      dispatch(success(reducerTypes.CREATE_PERMIT_CONDITION));
+      return response;
+    })
+    .catch(() => dispatch(error(reducerTypes.CREATE_PERMIT_CONDITION)))
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const deletePermitCondition = (permitAmdendmentGuid, permitConditionGuid) => (dispatch) => {
```

Comment:
same comment on types.

Reply:
Same here, it wasn't tied to MinePermitInfo so left it as it is for now